### ISSUE TITLE
Don't backprop on character probabilities for inputs where the number plate isn't present

### ISSUE THREAD

**Issue by matthewearl** [matthewearl/deep-anpr#7]
Training data contains images where the number plate is too large/too small/not present/truncated.  In this case the network should output "0" for the presence parameter, and we don't care what the character outputs are.  Currently train.py applies back-propagation to the character outputs for inputs where the plate is not present (in the sense described above).  Training could be made more efficient by only backpropping on the presence output in these cases.

**Comment by synckey** [comment]
The code multiplies the presence_loss with 10.0, so I think the presence loss dominants. If you want to ignore the digits_loss, you can   change this line and set the none presence ones to zero .
```
  digits_loss = tf.nn.softmax_cross_entropy_with_logits(
                                          tf.reshape(y[:, 1:],
                                                     [-1, len(common.CHARS)]),
                                          tf.reshape(y_[:, 1:],
                                                     [-1, len(common.CHARS)]))
```

**Comment by matthewearl** [comment]
I've got a fix ready which zeros the `digits_loss` entries which correspond with non-present plates.  Biasing the `presence_loss` by a factor of ~10 still seems to be useful though - without it training diverges with the current learning rate/batch size parameters. Intuitively this makes sense as the maximum `presence_loss` is smaller that the maximum `digits_loss` by a factor of 7 but I'm not sure this is the full story.

I'll push the fix and close this issue once I've done more investigation.
